cpp/src/gandiva/greedy_dual_size_cache.h

```diff
@@ -101,12 +101,16 @@ class GreedyDualSizeCache {
       if (size() >= capacity_) {
         evict();
       }
+      // check and handle possible overflow
+      if (UINT64_MAX - value.cost < inflation_) {
+        overflow();
```

Comment:
You don't need to do it for every insert. You can do it inside evict since only there inflation is updated

Reply:
Even if the inflation is not updated, it is still possible that the inserted **value.cost + inflation** generates an overflow. This is why we need to always check it during insertion or get methods

Reply:
Since we are resetting here the current costs are lost. Set the new costs to current_cost - inflation